core/client/client.go

```diff
@@ -161,43 +159,47 @@ func (c *Client) ListHistory(ctx context.Context, userID, appID string, start, e
 	epochsReceived := int64(0)
 	epochsWant := end - start + 1
 	for epochsReceived < epochsWant {
-		trustedSnapshot := c.trusted
-		resp, err := c.cli.ListEntryHistory(ctx, &pb.ListEntryHistoryRequest{
-			DomainId:      c.domainID,
-			UserId:        userID,
-			AppId:         appID,
-			FirstTreeSize: int64(trustedSnapshot.TreeSize),
-			Start:         start,
-			PageSize:      min(int32((end-start)+1), pageSize),
-		}, opts...)
-		if err != nil {
-			return nil, err
-		}
-		epochsReceived += int64(len(resp.GetValues()))
-
-		for i, v := range resp.GetValues() {
-			Vlog.Printf("Processing entry for %v, epoch %v", userID, start+int64(i))
-			smr, slr, err := c.VerifyGetEntryResponse(ctx, c.domainID, appID, userID, trustedSnapshot, v)
+		{
```

Comment:
Do you need this pair of braces? I believe the deferred `Unlock` is called after each iteration anyway, isn't it?

Reply:
Oh, I was wrong, sorry. Apparently it is called after the loop exits.